server/coordinator/procedure/storage.go

```diff
@@ -0,0 +1,22 @@
+package procedure
+
+import (
+	"context"
+)
+
+type Write interface {
+	CreateOrUpdate(ctx context.Context, meta *Meta) error
+}
+
+// nolint
+type Meta struct {
+	id      uint64
+	typ     procedureType
+	state   State
+	rawData []byte
+}
+
+type Storage interface {
+	Write
+	List(ctx context.Context, state State) (*Meta, error)
```

Comment:
- The `state` parameter maybe prevents callers to fetch all the Procedure Meta from the storage.
- Should the returned value be `[]*Meta`?

BTW, maybe we should revise the method signature to something like https://github.com/CeresDB/ceresmeta/blob/5eb015274de053acc027ce5cec5f22de795b60af/server/etcdutil/util.go#L27